File: public_web/tests/vixrex-turkce-morfoloji-parite.test.ts
import { describe, expect, it } from "vitest";
import { resolveVixrexIntent, resolveVixrexIntentsAll } from "../src/lib/vixrexIntentResolver";
import { extractVixrexValue } from "../src/lib/vixrexValueExtractor";

const cases = [
  ["Dükkanın adı Aymira Giyim", "isletmeAdi", "Aymira Giyim"],
  ["Dükkanımın adı Aymira Giyim", "isletmeAdi", "Aymira Giyim"],
  ["Ürünlerin başlığı Ürünlerimiz", "urunBolumBaslik", "Ürünlerimiz"],
  ["Galerinin başlığı Yaptığımız İşler", "galeriBaslik", "Yaptığımız İşler"],
] as const;

describe("Türkçe iyelik/genitif — Next.js", () => {
  for (const [input, anahtar, deger] of cases) {
    it(`${input} -> ${anahtar}`, () => {
      const alan = resolveVixrexIntent(input);
      expect(alan?.anahtar).toBe(anahtar);
      expect(alan ? extractVixrexValue(input, alan) : null).toBe(deger);
    });
  }

  it("kısa 'il' aliası normal kelime içinden veya ek genişlemesinden doğmaz", () => {
    for (const input of [
      "Ailece müşterilerimize hizmet veriyoruz",
      "İlgili bilgiyi sonra ekleriz",
      "Kaliteli hizmet veriyoruz",
    ]) {
      expect(resolveVixrexIntentsAll(input).map((a) => a.anahtar)).not.toContain("il");
    }
  });
});

File: test/vixrex_turkce_morfoloji_parite_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:vixrex/services/vixrex_nlu/vixrex_intent_resolver.dart';
import 'package:vixrex/services/vixrex_nlu/vixrex_value_extractor.dart';

void main() {
  const resolver = VixrexIntentResolver();
  const extractor = VixrexValueExtractor();

  final cases = <({String input, String anahtar, String deger})>[
    (input: 'Dükkanın adı Aymira Giyim', anahtar: 'isletmeAdi', deger: 'Aymira Giyim'),
    (input: 'Dükkanımın adı Aymira Giyim', anahtar: 'isletmeAdi', deger: 'Aymira Giyim'),
    (input: 'Ürünlerin başlığı Ürünlerimiz', anahtar: 'urunBolumBaslik', deger: 'Ürünlerimiz'),
    (input: 'Galerinin başlığı Yaptığımız İşler', anahtar: 'galeriBaslik', deger: 'Yaptığımız İşler'),
  ];

  group('Türkçe iyelik/genitif — Flutter', () {
    for (final c in cases) {
      test('${c.input} -> ${c.anahtar}', () {
        final alan = resolver.resolve(c.input);
        expect(alan?.anahtar, c.anahtar);
        expect(alan == null ? null : extractor.extract(c.input, alan), c.deger);
      });
    }

    test("kısa 'il' aliası normal kelime içinden doğmaz", () {
      for (final input in [
        'Ailece müşterilerimize hizmet veriyoruz',
        'İlgili bilgiyi sonra ekleriz',
        'Kaliteli hizmet veriyoruz',
      ]) {
        expect(
          resolver.resolveAll(input).map((a) => a.anahtar),
          isNot(contains('il')),
        );
      }
    });
  });
}
